client/lib/route/path.ts

```diff
@@ -58,9 +58,12 @@ export function getSiteFragment( path: URLString ): SiteSlug | SiteId | false {
 
 	// Check last and second-to-last piece for numeric site ID
 	for ( let i = 2; i > 0; i-- ) {
-		const piece = parseInt( pieces[ pieces.length - i ], 10 );
-		if ( Number.isSafeInteger( piece ) ) {
-			return piece;
+		const piece = pieces[ pieces.length - i ];
+		// We can't just do parseInt because some string look like numbers, eg: '404-hello'
+		const isNumber = String( piece ).match( /\D/ ) === null;
```

Comment:
I think the following (equivalent in terms of the result) syntax conveys the intent a bit better:
```js
const isNumber = /^\d+$/.test( piece );
```
`RegExp.test` returns a boolean instead of the full match structure we don't need, and converts `piece` to a string automagically.

Reply:
Perfect. Thanks!